presto-memory-context/src/main/java/com/facebook/presto/memory/LocalMemoryContext.java

```diff
@@ -19,17 +19,22 @@
 import javax.annotation.concurrent.GuardedBy;
 import javax.annotation.concurrent.ThreadSafe;
 
+import java.io.Closeable;
+
 import static com.facebook.presto.memory.MemoryReservationHandler.NOT_BLOCKED;
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static java.util.Objects.requireNonNull;
 
 @ThreadSafe
 public class LocalMemoryContext
+    implements Closeable
```

Comment:
I personally don't like it when classes implements `Closeable`. This makes it really hard to find out the callers of the method. I prefer not implementing `Closeable` unless there is a particular usability benefit. Since `Closer.register` takes something like `x::close`, I rarely find usability benefits given we don't use try-with-resource (since the life cycle of this class is beyond a single method).